rpcserver.go

```diff
@@ -3499,3 +3512,25 @@ func (r *rpcServer) RemoveUTXOLease(ctx context.Context,
 
 	return &wrpc.RemoveUTXOLeaseResponse{}, nil
 }
+
+// marshalMetaType map the native meta type to the RPC type.
+func marshalMetaType(metaType proof.MetaType) (taprpc.AssetMetaType, error) {
+	switch metaType {
+	case proof.MetaOpaque:
+		return taprpc.AssetMetaType_META_TYPE_OPAQUE, nil
+
+	default:
+		return 0, fmt.Errorf("unknown meta type: %d", metaType)
+	}
+}
+
+// unmarshalMetaType maps an RPC meta type into a concrete type.
+func unmarshalMetaType(rpcMeta taprpc.AssetMetaType) (proof.MetaType, error) {
+	switch rpcMeta {
+	case taprpc.AssetMetaType_META_TYPE_OPAQUE:
+		return proof.MetaOpaque, nil
+
+	default:
+		return 0, fmt.Errorf("unknown meta type: %d", rpcMeta)
```

Comment:
Similar here re allowing the unknown meta types to be encoded/decoded.

Reply:
Okay, I think I'm going to do what I suggested [above](https://github.com/lightninglabs/taproot-assets/pull/525#discussion_r1338527288) and see how that looks.